azurelinuxagent/common/osutil/default.py

```diff
@@ -390,12 +390,12 @@ def conf_sudoer(self, username, nopasswd=False, remove=False):
         if not remove:
             # for older distros create sudoers.d
             if not os.path.isdir(sudoers_dir):
-                sudoers_file = os.path.join(sudoers_dir, '../sudoers')
                 # create the sudoers.d directory
-                os.mkdir(sudoers_dir)
+                fileutil.mkdir(sudoers_dir)
```

Comment:
The sudoers needs to be created with specific protections. Please specify `mode=0o550`. Just as /etc/sudoers itself should be mode 0440 (not writeable), the /etc/sudoers.d directory shouldn't be writeable. (Root can write anything, of course.)

Reply:
True, but the missing permissions are not introduced with this PR. Merging as it still fixes an issue with the code.